WordPress/src/main/java/org/wordpress/android/ui/sitecreation/verticals/SiteCreationIntentsFragment.kt

```diff
@@ -0,0 +1,88 @@
+package org.wordpress.android.ui.sitecreation.verticals
+
+import android.content.Context
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.view.isInvisible
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModelProvider
+import org.wordpress.android.R
+import org.wordpress.android.WordPress
+import org.wordpress.android.databinding.SiteCreationIntentsFragmentBinding
+import org.wordpress.android.ui.utils.UiHelpers
+import org.wordpress.android.util.DisplayUtilsWrapper
+import javax.inject.Inject
+
+/**
+ * Implements the Site Intent Question UI
+ */
+@Suppress("TooManyFunctions")
+class SiteCreationIntentsFragment : Fragment() {
+    @Inject internal lateinit var viewModelFactory: ViewModelProvider.Factory
+    @Inject internal lateinit var uiHelper: UiHelpers
+    @Inject internal lateinit var displayUtils: DisplayUtilsWrapper
+
+    private lateinit var viewModel: SiteCreationIntentsViewModel
+    private var binding: SiteCreationIntentsFragmentBinding? = null
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        (requireActivity().applicationContext as WordPress).component().inject(this)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.site_creation_intents_fragment, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        viewModel = ViewModelProvider(requireActivity(), viewModelFactory)
+                .get(SiteCreationIntentsViewModel::class.java)
+
+        val binding = SiteCreationIntentsFragmentBinding.bind(view)
+        with(binding) {
+            this@SiteCreationIntentsFragment.binding = binding
+            setupUi()
+            setupViewModel()
+            setupActionListeners()
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        (requireActivity().application as WordPress).component().inject(this)
```

Comment:
Do we still need this if we are injecting in `onAttach`?

Reply:
Nope, good observation!
Many thanks 👍, I'll review the code to make sure we're only doing it once :)

Reply:
✅ Done